src/ndarray/ndarray_function.cc

```diff
@@ -183,5 +183,13 @@ void ElementwiseSum<cpu>(mshadow::Stream<cpu>* s,
   }
 }
 
+
+template<>
+void SetValueRsp<cpu>(mshadow::Stream<cpu> *s,
```

Comment:
Would it be better for `ndarray.cc` to call `SetValue(stream, real_t, NDArray)`, then `SetValue` internally checks the stype and calls `SetValueRsp`?